src/openrct2/world/Scenery.h

```diff
@@ -191,8 +191,7 @@ struct rct_scenery_group_entry
 {
     rct_string_id name;
     uint32_t image;
-    ScenerySelection scenery_entries[0x80];
-    uint8_t entry_count;
+    std::vector<ScenerySelection> SceneryEntries;
     uint8_t priority;
     uint32_t entertainer_costumes;
 };
```

Comment:
This structure should be removed altogether. Everything should access the `SceneryGroupObject` directly. There probably aren't too many references to change for scenery groups.

Reply:
That would mean redesigning how objects work to remove the GetLegacy function which would be great to do but will take a lot of work as all the other objects need the same fixes.